HW_8/task 8.2.js

```diff
@@ -0,0 +1,269 @@
+// Task 2. Перед вами структура компании, и ниже представлены задания, относящиеся к ней.
+// В заданиях по максимуму использовать методы массивов, создавать функции-помощники, выполняющие дополнительные действия,
+// чтобы ваши функции выполняли строго одну работу. ЭТО ОЧЕНЬ ВАЖНО!
+
+const enterprises = [
+  {
+    id: 1,
+    name: 'Предприятие 1',
+    departments: [
+      {
+        id: 2,
+        name: 'Отдел тестирования',
+        employees_count: 10,
+      },
+      {
+        id: 3,
+        name: 'Отдел маркетинга',
+        employees_count: 20,
+      },
+      {
+        id: 4,
+        name: 'Администрация',
+        employees_count: 15,
+      },
+    ],
+  },
+  {
+    id: 5,
+    name: 'Предприятие 2',
+    departments: [
+      {
+        id: 6,
+        name: 'Отдел разработки',
+        employees_count: 50,
+      },
+      {
+        id: 7,
+        name: 'Отдел маркетинга',
+        employees_count: 20,
+      },
+      {
+        id: 8,
+        name: 'Отдел охраны труда',
+        employees_count: 5,
+      },
+    ],
+  },
+  {
+    id: 9,
+    name: 'Предприятие 3',
+    departments: [
+      {
+        id: 10,
+        name: 'Отдел аналитики',
+        employees_count: 0,
+      },
+    ],
+  },
+];
+
+// Задания:
+// 1. Вывести все предприятия и их отделы. Рядом указать количество сотрудников.
+// Для предприятия посчитать сумму всех сотрудников во всех отделах.
+
+// **Пример:**
+
+// Предприятие 1 (45 сотрудников)
+// - Отдел тестирования (10 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Администрация (15 человек)
+// Предприятие 2 (75 сотрудников)
+// - Отдел разработки (50 сотрудников)
+// - Отдел маркетинга (20 сотрудников)
+// - Отдел охраны труда (5 сотрудников)
+// Предприятие 3 (нет сотрудников)
+// - Отдел аналитики (нет сотрудников)
+
+const enterpriseStructure = (enterprises) => {
+    enterprises.map(enterprise => {
+        const totalEmployees = enterprise.departments.reduce((acc, department) => acc + department.employees_count, 0);
+        if (/.*1$/.test(totalEmployees) && !/.*11$/.test(totalEmployees)) {
+            console.log(`${enterprise.name} (${totalEmployees} сотрудник)`);
```

Comment:
You have repeated lines, you can move this logic outside and reuse
```js
const isEndsWithOne = (count) => /.*1$/.test(count) && !/.*11$/.test(count);

if (isEndsWithOne(totalEmployees)) {} ....
```

Reply:
Thanks, will do it